develop/tutorials/articles/193-adaptive-media/00-using-adaptive-media-apis-intro.markdown

```diff
@@ -0,0 +1,283 @@
+example, the return type is `Option<Integer>`.
+
+### Obtaining and searching Adaptive Media Images
+
+The Adaptive Media API provides a way to search and obtain Adaptive Media by 
+providing a query with the search parameters. The
+[`com.liferay.adaptive.media.finder.AdaptiveMediaFinder` interface](https://github.com/liferay/com-liferay-adaptive-media/blob/master/adaptive-media-api/src/main/java/com/liferay/adaptive/media/finder/AdaptiveMediaFinder.java) 
+provides the APIs needed for this search.
+
+If the search is specific to Adaptive Images, you can use the 
+[`com.liferay.adaptive.media.image.finder.ImageAdaptiveMediaFinder` interface](https://github.com/liferay/com-liferay-adaptive-media/blob/master/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/finder/ImageAdaptiveMediaFinder.java) 
+to obtain a list of adaptive images.
+
+This interface is implemented by the 
+[`com.liferay.adaptive.media.image.internal.finder.ImageAdaptiveMediaFinderImpl` class](https://github.com/liferay/com-liferay-adaptive-media/blob/master/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/finder/ImageAdaptiveMediaFinderImpl.java) 
+and is registered as an OSGi component using the 
+[`ImageAdaptiveMediaFinder` interface](https://github.com/liferay/com-liferay-adaptive-media/blob/master/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/finder/ImageAdaptiveMediaFinder.java) 
+and the property 
+`model.class.name=com.liferay.portal.kernel.repository.model.FileVersion`. This 
+property specifies the kind of model the finder expects. All current 
+implementations expect `FileVersion` models.
+
+You can obtain the `ImageAdaptiveMediaFinder` OSGI component using Declarative 
+Services:
+
+    @Reference(
+    
+      target =
+    "(model.class.name=com.liferay.portal.kernel.repository.model.FileVersion)"
+    
+    )
+    
+    private ImageAdaptiveMediaFinder _finder;
+
+The finder accepts a function that, when invoked, receives a `queryBuilder` that
+specifies the criteria for the search. The methods accepted by the image 
+`queryBuilder` can be seen in the 
+[`com.liferay.adaptive.media.image.finder.ImageAdaptiveMediaQueryBuilder` interface](https://github.com/liferay/com-liferay-adaptive-media/blob/master/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/finder/ImageAdaptiveMediaQueryBuilder.java). 
+The `queryBuilder` allows you to run powerful queries, as shown below:
+
+    Stream<AdaptiveMedia<ImageAdaptiveMediaProcessor>> stream =
+    
+      _finder.getAdaptiveMedia(
+    
+         queryBuilder ->
+    
+            queryBuilder.allForFileEntry(fileEntry).
+    
+            orderBy(ImageAdaptiveMediaAttribute.IMAGE_WIDTH, true).
+    
+            done());
+
+This returns a stream of the image variants created for the image fileEntry 
+in ascending order by width.
+
+<!-- Is this using an InputStream to get the content or is this completely 
```

Comment:
Is this using an InputStream to get the content or is this completely different?

    Stream<AdaptiveMedia<ImageAdaptiveMediaProcessor>> stream =
    
      _finder.getAdaptiveMedia(
    
         queryBuilder ->
    
            queryBuilder.
    
               forVersion(_fileVersion).
    
               with(ImageAdaptiveMediaAttribute.IMAGE_HEIGHT, 100).
    
               done());